nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -940,7 +976,7 @@ def from_tensorflow(self, graph):
                         "Please freeze the graph with add_shapes=True")
             elif node.op == "Const":
                 # Assuming first Const node as Graph Input node
-                if self._input_node == '':
+                if self._input_node == '' and not has_placeholders:
```

Comment:
Technically there is nothing does with _input_node. Hence _input_node has no significance here.
As you touched this area I suggest to change the below or ignore this part in this PR.

A graph can be executed as ConstA+ConstB=C.
where there is no place holders and we don't need to convert an Const to placeholder and expect input from user (Example test cases.).

Hence I suggest to simplify as "All placeholders are inputs" and "all Constants are Params".

Reply:
Treating constants as inputs seemed strange to me too, so I am glad to see you suggestion. 

I attempted to make this simplification and it seems that at least some operations implicitly depend on this input constants. For example, `_conv` operation uses `_input_shape` [here](https://github.com/dmlc/tvm/blob/master/nnvm/python/nnvm/frontend/tensorflow.py#L181) . Input shape in its turn is set only for input nodes (i.e. nodes which aren't listed in `params` attribute, [here is the key point](https://github.com/dmlc/tvm/blob/cd0dda25970e235696d4ab210432a9f865bb9606/nnvm/python/nnvm/frontend/tensorflow.py#L996)) Thus, removing constants from inputs unfortunately breaks some tests involving convolution, since `_input_shapes` is no longer available for input[0].

I think it is a bug in `_conv` since any operation should AFAIK work with any type of input nodes. Let me suggest that I fix it in a separate commit. As for this commit, I'd better remove the placeholder tweak completely to let other changes go.

Reply:
Done, postponing the placeholders part. I've started https://discuss.tvm.ai/t/importing-tensorflow-models-placeholder-vs-const-issues/561 topic for futher discussion.